cpp/CMakeLists.txt

```diff
@@ -205,6 +205,7 @@ add_library(cugraph SHARED
     src/visitors/graph_envelope.cpp
     src/visitors/visitors_factory.cpp
     src/visitors/bfs_visitor.cpp
+    src/visitors/rw_visitor.cpp
```

Comment:
So, do we plan to place bindings related files under the cugrpah/cpp directory? Or we'd better separate the core libcugraph from the bindings related files? (i.e. placing bindings related files under something like cugraph/bindings)

Reply:
The visitor implementation is not restricted to bindings only. It's meant to be (and it is) used for implementation purposes, too. These _are_ cpp files. There will be a separate branch for the C-only API.

Reply:
Then, what about creating a separate directory under the cpp (similar to tests)? Is this directly relevant to libcugraph users? (someone willing to directly use our C++ API). If not, I am not sure it is a good idea to include this in our core libcugraph_(sg|mg).so.

Reply:
Yes, it is. As you probably know Visitor is useful to enrich the functionality of an object (visitee) without having to add new methods to it, which may inflate the code and cause merge conflicts on high traffic headers, like `graph*.hpp` .

The visitor abstraction is being used in RW, for example, for exactly such reasons.

Also, this structuring of this code has been in place for a while. And it has been extensively discussed before-hand (at least 2 presentations and demos have been done). Constantly shuffling code around _after the fact_ is counterproductive, confusing, and wasteful. If new code restructuring is in the works, then this is _not_ the PR for it. This PR has different goals. We've already agreed on small, targeted PRs as opposed to all-inclusive major code restructuring ones.

There's also the question of what exactly is the criterion for code cut-off to be accepted in `cugraph`. If this is to be discussed, then, again, maybe this is something to discuss _before-fact_, not after. I am starting to get the feeling that it's increasingly hard to get code accepted in `cugraph` and a lot of effort is potentially wasted with code shuffling and reshuffling on what appears to be quite ad-hoc and last-moment criteria.